### DISCUSSION TITLE
linestyle.light - lines break on RaspberryPiOS

### DISCUSSION THREAD

**Discussion by ETC-kintaka** [mum4k/termdash#333] [Show and tell]
Hello everyone.
I have tried running termdash on a RaspberryPiOS.
As a result, the ruled lines are broken.

[broken png file.](https://cdn-ak.f.st-hatena.com/images/fotolife/k/kamuycikap/20220729/20220729140051.png)

Am I the only one with this symptom?
Am I screwing up my termdash setup?

-------
No LSB modules are available.
Distributor ID:	Raspbian
Description:	Raspbian GNU/Linux 11 (bullseye)
Release:	11
Codename:	bullseye

**Comment by mum4k** [comment]
Hi @ETC-kintaka, this class of problems usually occurs due to using an incompatible terminal emulator. I have to say I did not try running this on a RaspberryPiOS.

Termdash uses [tcell](https://github.com/gdamore/tcell) or [termbox](https://github.com/nsf/termbox) as its terminal drivers. As such, it only supports what these libraries support. You could try [switching](https://github.com/mum4k/termdash/blob/9954a0543ed738c7ccb78ecd8eaed8e21db6ae7e/widgets/text/textdemo/textdemo.go#L71) between them, but `tcell` generally is the more compatible one.

Can you confirm what terminal emulator you are using? I would also recommend looking at its settings to see if there are any options that could help here.

**Comment by ETC-kintaka** [comment]
Hello. num4k
Thank you very much.

I am a little late in replying.
I live in Japan.

I took your advice.
I don't understand tcell correctly, but I understood that it is a problem with Terminal settings.

I ran the sample program on UbuntuLinux 20.04.2 as well.
I also ran the sample program on a Windows 11 terminal.
The result was the same.

The version of the UbuntuLinux terminal is GnomeShell 3.36.7.
I changed the terminal options.
I set the character encoding to UTF8.
Setting the ambiguous width characters to double-byte characters in the compatibility settings improves things a bit.

I am considering using a compatible terminal, but could you please let me know which terminal works reliably?

I tried manipulating the settings, but the result was not resolved.
Is it because I set up Ubuntu in Japanese?
Is it a character font issue?

**Comment by ETC-kintaka** [comment]
I will add my findings.
I write from the conclusion.

I have rewritten tcell to termbox as follows.
Then, it is now displayed without any problem.
However, I could not realize the italic specification of the text because of a compile error.

I am Japanese.
I set up a Japanese Linux system.
Ubuntu 20.40
MXLinux 21.1
RaspberryPi OS 11 (bullseye)
I have tried all the above Linux systems with standard Terminal, but tcell seems to be incompatible.
Is the Japanese font affected?

--- before
func main() {
	t, err := tcell.New() // <-- before
	if err ! = nil {
		panic(err)
	}
	defer t.Close()


--- after
func main() {
    t, err := termbox.New() // <-- after
    if err ! = nil {
        panic(err)
    }
    defer t.Close()

**Reply by mum4k** [reply]
Thank you for finding out that this problem seems to be limited to `tcell`. I would suggest trying to reproduce this problem with `tcell` directly (without `termdash` in the mix). If you manage to reproduce, you could open an issue with the `tcell` repository.
